### PR TITLE
haxe.io.Bytes documentation.

### PR DESCRIPTION
**Pull request by cancerberoSgx**
As a new user it took me some time to figure out what undocumented methods of haxe.io.Bytes do. Knowing exactly the behavior or this class is important for performance and with simple comments like these it should be enough. Thanks.

### CODE REVIEW

File: std/haxe/io/Bytes.hx [hunk 1]
```diff
@@ -73,6 +79,9 @@ class Bytes {
 		#end
 	}
 
+	/**
+	  Copies `len` bytes from `src` into this instance.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: What about the other arguments? `len` bytes are copied, that's true. The destination is `pos` bytes into `this` instance, and the source is `srcpos` bytes into `src`.

File: std/haxe/io/Bytes.hx [hunk 2]
```diff
@@ -111,6 +120,9 @@ class Bytes {
 		#end
 	}
 
+	/**
+	  Sets the value of this instance's `len` bytes to `value`, starting from index `pos`.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: "Sets `len` consecutive bytes starting from index `pos` of `this` instance to `value`."

File: std/haxe/io/Bytes.hx [hunk 3]
```diff
@@ -130,6 +142,9 @@ class Bytes {
 		#end
 	}
 
+	/**
+	  Returns a new `Bytes` instance that contains a copy of this instance's `len` bytes, starting at index `pos`.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: "… that contains a copy of `len` bytes of `this` instance, starting at index `pos`."

File: std/haxe/io/Bytes.hx [hunk 4]
```diff
@@ -156,7 +171,11 @@ class Bytes {
 		return new Bytes(len, b.slice(pos, pos + len));
 		#end
 	}
-
+	
+	/**
+	  Returns `0` if this instance's bytes and given `other`'s bytes are identical, or a value
+	  different than `0` otherwise.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: Non-zero values also have a meaning, it should be lexicographic comparison (negative value if `this` is shorter or same length + lexicographically comes first, positive if `this` is longer or same length + lexicographically comes later second).

**Review by Aurel300 — CHANGES_REQUESTED** [review]

File: std/haxe/io/Bytes.hx [hunk 5]
```diff
@@ -81,6 +81,10 @@ class Bytes {
 
 	/**
 	  Copies `len` bytes from `src` into this instance. 
+    @param pos Zero-based location in this instance from which to start copying bytes.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: `from which to start copying` -> `at which to start writing`

File: std/haxe/io/Bytes.hx [hunk 6]
```diff
@@ -121,7 +125,7 @@ class Bytes {
 	}
 
 	/**
-	  Sets the value of this instance's `len` bytes to `value`, starting from index `pos`.
+    Sets `len` consecutive bytes starting from index `pos` of this instance to `value`.
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: Use tabs for indentation, please.

File: std/haxe/io/Bytes.hx [hunk 7]
```diff
@@ -173,8 +177,14 @@ class Bytes {
 	}
 	
 	/**
-	  Returns `0` if this instance's bytes and given `other`'s bytes are identical, or a value
-	  different than `0` otherwise.
+	  Returns `0` if this instance's bytes and given `other`'s bytes are identical. 
+    
+    Otherwise, returns a negative value if this instance's `length` is less than `other`'s `length` or if the
```
Comment (Aurel300) [inline std/haxe/io/Bytes.hx]: Drop the `Otherwise,`. Use tabs in this doc as well.

**Review by Aurel300 — CHANGES_REQUESTED** [review]

**Comment by Simn** [comment]
@Aurel300 Can you finalize this PR?

**Review by Aurel300 — APPROVED** [review]

**Pull request merged**